### DISCUSSION TITLE
TransactionCanceledException: Transaction cancelled, please refer cancellation reasons for specific reasons [ConditionalCheckFailed, None]

### DISCUSSION THREAD

**Discussion by KristianLake** [sensedeep/dynamodb-onetable#278] [Q&A] [answered]
Hi, when executing the following code in a lambda function

```
  let user = await table.transact('write', transaction).catch((error) => {
    let errorMessage = 'Could not create user'

    console.log(error.getMessage())
    console.log(transaction.TransactItems)

    return {
      error: errorMessage
    }
  })
```
Assuming the transaction fails and has multiple transactions inside, it does error correctly

```
neTableError: OneTable execute failed "transactWrite" for "_Generic. Transaction cancelled, please refer cancellation reasons for specific reasons [ConditionalCheckFailed, None]
    at Table.execute 
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async Table.transact {
  context: {
    err: TransactionCanceledException: Transaction cancelled, please refer cancellation reasons for specific reasons [ConditionalCheckFailed, None]
```

You cant seem to get the more specific error message (cancellation reasons) though.

How do you get the specific error message

The pseudo code for this is

```
let errorMessage = 'foo'
if(error.cancellationreason[0].code === 'conditionalcheckfailed) {
errormessage = 'foo already exists'
} else if if(error.cancellationreason[1].code === 'conditionalcheckfailed) {
errormessage = 'bar already exists'
} 
```
Without being able to check which bit of the transaction failed you only know something in the transaction failed but not what.

thanks alot! :)

**Comment by mobsense** [comment, marked as answer]
With the AWS V3 SDK, you should see those details in

    err.context.CancellationReasons.

For the AWS V2 SDK, that info does not exist in the AWS V2 Javascript API. Only in V3.

See https://github.com/aws/aws-sdk-go/issues/2318

> Thank you for reaching out about this @guregu. While the CancellationReason structure is present in the API model for DynamoDB, the AWS SDK for Java is the only SDK (V2) that currently supports unmarshalling this structure. I've marked this issue as a Feature Request, and will bring it up in our next sprint.

**Comment by mobsense** [comment]
Transferring to discussions.
